Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/TranslationHandles.cs

```diff
@@ -0,0 +1,338 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Experimental.UI.BoundsControlTypes;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI.BoundsControl
+{
+    /// <summary>
+    /// Translation handles for <see cref="BoundsControl"/> that are used for translating the
+    /// Gameobject BoundsControl is attached to with near or far interaction
+    /// </summary>
+    public class TranslationHandles : HandlesBase
+    {
+        protected override HandlesBaseConfiguration BaseConfig => config;
+        protected TranslationHandlesConfiguration config;
+        protected FlattenModeType cachedFlattenAxis;
+
+        internal TranslationHandles(TranslationHandlesConfiguration configuration)
+        {
+            Debug.Assert(configuration != null, "Can't create BoundsControlTranslationHandles without valid configuration");
+            config = configuration;
+            config.handlesChanged.AddListener(HandlesChanged);
```

Comment:
Out of curiousity, does handlesChanged add a ref to this object?

i.e. it would be odd to me that this object be destroyed while it has an ongoing registration as a listener.

i.e. how does ~TranslationHandles get called when it's registered?

Reply:
@wiwei  I am not familiar with how our event listener system works with this stuff. This is directly copied from the other handles. I would defer to Berni for questions about that.

Reply:
I might need some more information on this as well - is the issue that the handles object dies while configuration might still send messages? handles unsubscribe from config event subscriptions when their dtor is called which happens when bounds control gets destroyed.

Reply:
I'm having some trouble finding the initial lines for this, but from what I remember was, my confusion was I thought that there might be a circular reference here:

i.e. when this object registers for an event, this I would assume causes a ref on this object to increment (otherwise, things would crash when the event invoked on a destroyed object).

However when I looked, I only saws that unregistration only occured in the dtor - meaning it would only happen once the object was destroyed. However, this seems impossible because the registration itself should increase the reference count (meaning, the act of registration == destructor won't be called).

This was my confusion, but I could have misread any of the parts within there.